src/display/editor/editor.js

```diff
@@ -746,6 +745,86 @@ class AnnotationEditor {
     }
 
     this.#hasBeenSelected = true;
+
+    this.#setUpDragSession(event);
+  }
+
+  #setUpDragSession(event) {
+    if (!this._isDraggable) {
+      return;
+    }
+
+    // Avoid to have spurious text selection in the text layer when dragging.
+    this._uiManager.enableUserSelect(false);
+
+    const savedParent = this.parent;
+    const savedX = this.x;
+    const savedY = this.y;
+
+    const pointerMoveOptions = { passive: true, capture: true };
+    const pointerMoveCallback = e => {
+      const [parentWidth, parentHeight] = this.parentDimensions;
+      const [tx, ty] = this.screenToPageTranslation(e.movementX, e.movementY);
+      this.x += tx / parentWidth;
+      this.y += ty / parentHeight;
+      if (this.x < 0 || this.x > 1 || this.y < 0 || this.y > 1) {
+        // The element will be outside of its parent so change the parent.
+        const { x, y } = this.div.getBoundingClientRect();
+        if (this.parent.findNewParent(this, x, y)) {
+          this.x -= Math.floor(this.x);
+          this.y -= Math.floor(this.y);
+        }
+      }
+
+      this.div.style.left = `${(100 * this.x).toFixed(2)}%`;
+      this.div.style.top = `${(100 * this.y).toFixed(2)}%`;
+      this.div.scrollIntoView({ block: "nearest" });
+    };
+    window.addEventListener(
+      "pointermove",
+      pointerMoveCallback,
+      pointerMoveOptions
+    );
+    window.addEventListener(
+      "pointerup",
+      e => {
+        this._uiManager.enableUserSelect(true);
```

Comment:
I've not tested it myself, but is there a risk that we could ever miss a "pointerup" event if the user e.g. ends the drag-operation *outside* of the current browser window, or switches to another program (e.g. with Alt+Tab on Windows) during dragging?

Because if so, we might accidentally leave e.g. text-selection disabled in the viewer.
To reduce the risk of problems happening elsewhere, can you please *also* add this line unconditionally when we disable editing?